main.js

```diff
@@ -0,0 +1,187 @@
+
+// **************************************** VARIABLES **************************************** //
+
+const posts = [];
+const $postButton = {};
+
+// **************************************** FUNCTIONS **************************************** //
+
+// Creates a post object, pushes to the posts array, and renders posts
+let postId = 0;
+const createPost = (author, message) => {
+  const post = {
+    author: author,
+    message: message,
+    votes: 0,
+    id: postId,
+    comments: [],
+    updatePost(newAuthor, newMessage) {
+      this.author = newAuthor;
+      this.message = newMessage;
+    },
+    addComment(comment) {
+      this.comments.push(comment);
+    },
+    deleteComment(comment) {
+      this.comments.splice(comments.indexOf(comment), 1);
+    }
+  };
+
+  // Add post to posts list and increment id for newly created posts (post.id)
+  posts.push(post);
+  postId++;
+};
+
+// Render HTML for posts
+const sortAndRenderPosts = () => {
+  posts.sort((a, b) => b.votes - a.votes);
+  
+  // Handlebars template
+  const source = $('#post-template').html();
+  const template = Handlebars.compile(source);
+
+  // Empty posts list, create HTML for each post, append to posts list
+  $(`.posts-list`).empty();
+  posts.map(post => {
+    let postHTML = template(post);
+    $(`.posts-list`).append(postHTML) ;
+  });
+};
+
+// Handles ability to upvote, downvote, edit, and delete posts, as well as add comments to posts using their respective buttons
+const handlePostButtonClick = jqButton => {
+  const $clickedButton = jqButton.buttonClicked;
+  const $associatedPost = jqButton.associatedPost;
+  const closestPostID = jqButton.associatedPost.data('id');
+
+  // Maps JS post obj to HTML post DOM element, execute appropriate button action, re-render posts
+  posts.map(post => {
+    if(post.id === closestPostID) {
+      // Upvote post
```

Comment:
With conditionals that extend beyond 2 or 3 conditions i'd consider using a switch statement for syntactic concision